crates/bevy_state/src/state/mod.rs

```diff
@@ -619,4 +619,84 @@ mod tests {
             1
         );
     }
+
+    #[derive(Resource, Default, Debug)]
+    struct TransitionTracker(Vec<&'static str>);
+
```

Comment:
For this test, I think it'd be ideal to have the computed state depend on both the main & sub states, so that there is no ambiguity in the desired scheduling. So I'd recommend adding a computed state for this:

```suggestion


    #[derive(PartialEq, Eq, Debug, Hash, Clone)]
    enum TransitionTestingComputedState {
        IsA,
        IsBAndEven,
        IsBAndOdd
    }

    impl ComputedStates for TransitionTestingComputedState {
        type SourceStates = (Option<SimpleState>, Option<SubState>);

        fn compute(sources: (Option<SimpleState>, Option<SubState>)) -> Option<Self> {
            match sources {
                (Some(simple), sub) => {
                    if simple == SimpleState::A
                    {
                        Some(Self::IsA)
                    } else if sub == Some(SubState::One) {
                        Some(Self::IsBAndOdd)
                    } else if sub == Some(SubState::Two) {
                        Some(Self::IsBAndEven)
                    } else {
                        None
                    }
                }
                _ => None,
            }
        }
    }
```